packages/contracts/helpers/init-helpers.ts

```diff
@@ -224,6 +226,199 @@ export const initAssetData = async (
   }
 };
 
+// adds more assets to an existing asset mappings, based on what is already in the asset mappings
+export const addAssetData = async (
+  reservesParams: iMultiPoolsAssets<IReserveParams>,
+  tokenAddresses: { [symbol: string]: tEthereumAddress },
+  verify: boolean,
+  curveMetadata?: iMultiPoolsAssets<ICurveMetadata>,
+  beethovenMetadata?: iMultiPoolsAssets<IBeethovenMetadata>
+) => {
+  // initTrancheMultiplier();
+  const addressProvider = await getLendingPoolAddressesProvider();
+
+  // Initialize variables for future reserves initialization
+  let reserveSymbols: string[] = [];
+
+  let underlying: string[] = [];
+
+  let initInputParams: {
+    asset: string;
+    defaultInterestRateStrategyAddress: string;
+    supplyCap: BigNumberish; //1,000,000
+    borrowCap: BigNumberish; //1,000,000
+    baseLTV: BigNumberish;
+    liquidationThreshold: BigNumberish;
+    liquidationBonus: BigNumberish;
+    borrowFactor: BigNumberish;
+    borrowingEnabled: boolean;
+    assetType: BigNumberish;
+    VMEXReserveFactor: BigNumberish;
+    tokenSymbol: string;
+  }[] = [];
+
+  let strategyRates: [
+    string, // addresses provider
+    string,
+    string,
+    string,
+    string
+  ];
+  let rateStrategies: Record<string, typeof strategyRates> = {};
+  let strategyAddresses: Record<string, tEthereumAddress> = {};
+
+  const reserves = Object.entries(reservesParams);
+
+  // Deploy init reserves per tranche
+  // tranche CONFIGURATION
+  const assetMappings = await getAssetMappings();
+
+  for (let [symbol, params] of reserves) {
+    if (!tokenAddresses[symbol]) {
+      console.log(
+        `- Skipping init of ${symbol} due token address is not set at markets config`
+      );
+      continue;
+    }
+    
+    const {
+      strategy,
+      assetType,
+      supplyCap, //1,000,000
+      borrowCap,
+      reserveDecimals,
+      baseLTVAsCollateral,
+      borrowFactor,
+      liquidationBonus,
+      liquidationThreshold,
+      borrowingEnabled,
+      reserveFactor,
+    } = params;
+
+    const {
+      optimalUtilizationRate,
+      baseVariableBorrowRate,
+      variableRateSlope1,
+      variableRateSlope2,
+    } = strategy;
+
+    const isAssetInMappings = await assetMappings.isAssetInMappings(tokenAddresses[symbol])
+    
+    if(isAssetInMappings) {
+      console.log("already have ", symbol, " in mappings. skipping")
+      const existingMapping = await assetMappings.getAssetMapping(tokenAddresses[symbol])
+      strategyAddresses[strategy.name] = existingMapping.defaultInterestRateStrategyAddress
+      continue
+    }
+    console.log("Attempting to process new asset: ", symbol)
+
+    const dbEntry = await getDbEntry(strategy.name);
+    if(dbEntry){
+      console.log("db already has ", strategy.name, " as ",dbEntry.address)
+      strategyAddresses[strategy.name] = dbEntry.address
+    }
+
+    if (!strategyAddresses[strategy.name]) {
```

Comment:
This section on creating and adding the strategy doesn't make sense to be inside the addAssetData monolith function. You should split this function up into the different components.

deployRateStrategy()
addAssetData()
Probably split up the addCurveMetadata and beethoven metadata steps as well